utils/text_formatting.tsx

```diff
@@ -636,6 +612,8 @@ function highlightCurrentMentions(
     return output;
 }
 
+const hashtagRegex = /(^|\W)(#\p{L}[\p{L}\d_.-]*[\p{L}\d])/gu;
```

Comment:
Same missing `\\` and order change as before.

Reply:
same as https://github.com/mattermost/mattermost-webapp/pull/9805#discussion_r816675976